### PR TITLE
[LF] [116326743] env + version should not be memoized

### PR DESCRIPTION
**Pull request by lfender6445**
https://www.pivotaltracker.com/story/show/116326743

Prevents version from being cached, which was borking the route lookup

NOTE i did not look at other strategies to see if the issue existed there. this just fixes header strat

### CODE REVIEW

File: spec/strategies/header_strategy_spec.rb [hunk 1]
```diff
@@ -2,17 +2,19 @@
...
       end
 
-      subject { described_class.new(header: 'Sue') }
+      subject { described_class.new(header: 'version') }
 
-      it "returns with v1" do
+      it "returns correct version" do
         expect(subject.get_version(env)).to eql("v1")
+        env['HTTP_VERSION'] = 'v2'
+        expect(subject.get_version(env)).to eql("v2")
```
Comment (lfender6445) [inline spec/strategies/header_strategy_spec.rb]: these tests must be run together so subject instance is cached (mimics config.ru which only creates on instance of the header strategy)

**Comment by pasha** [comment]
:+1:

**Comment by colinrymer** [comment]
How does this fit in with the decision that APIs should be versioned through the `Accept` header?

**Comment by lfender6445** [comment]
i wasn't a part of that conversation. i will say that phenotype lets you call it whatever you want - in the spec, i just call it version. @wflann did you have any feedback regarding colins comments

**Comment by colinrymer** [comment]
This doesn't properly parse the Accept header for the version though.

**Comment by lfender6445** [comment]
colin, that is a separate issue. if phenotype needs to respect accept header we can open another story for it.

**Comment by wshatch** [comment]
:+1: Accept header will need a new strategy.

**Pull request merged**